test/test_type_promotion.py

```diff
@@ -808,6 +808,80 @@ def test_integer_addcdiv_deprecated(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, '^Integer division.+is no longer supported+'):
             t.addcdiv_(t, t)
 
+    def test_addcdiv_promotion(self, device):
```

Comment:
Elementwise ternary tests might be interesting but this PR doesn't have to add them. Spot checking addcdiv and addcmul type promotion like this seems OK. I would also add a long x bfloat16 x float32 case.

Reply:
added the types you suggested

Reply:
From the test failures it looks like this test will need to be skipped on XLA and an issue filed with PyTorch/XLA. (A comment explaining the skip should include a link to the issue.)